src/main/java/com/splunk/hecclient/Indexer.java

```diff
@@ -156,16 +158,13 @@ public boolean send(final EventBatch batch) {
     public synchronized String executeHttpRequest(final HttpUriRequest req) {
         CloseableHttpResponse resp;
         if (hecConfig.kerberosAuthEnabled()) {
-            if (config == null) {
-                defineKerberosConfigs();
-            }
-            Set<Principal> princ = new HashSet<Principal>(1);
-            princ.add(new KerberosPrincipal(hecConfig.kerberosUser()));
-            Subject sub = new Subject(false, princ, new HashSet<Object>(), new HashSet<Object>());
             try {
-                LoginContext lc = new LoginContext("", sub, null, config);
-                lc.login();
-                Subject serviceSubject = lc.getSubject();
+                if (config == null) {
+                    defineKerberosConfigs();
+                }
+                if(serviceSubject != null){
+                    serviceSubject = getSubject();
```

Comment:
Don't know about the exact logic here but seems like logic should be like, if `serviceSubject` is null then we will get a new subject

Reply:
```suggestion
                if(serviceSubject == null){
                    serviceSubject = getSubject();
```